mlrun/builder.py

```diff
@@ -327,14 +328,19 @@ def build_image(
         requirements_list = None
         requirements_path = requirements
 
+    commands = commands or []
     if with_mlrun:
-        commands = commands or []
         mlrun_command = resolve_mlrun_install_command(
-            mlrun_version_specifier, client_version
+            mlrun_version_specifier, client_version, commands
         )
-        if mlrun_command not in commands:
+        if mlrun_command:
             commands.append(mlrun_command)
 
+        # mlrun prerequisite - upgrade pip
+        upgrade_pip_command = resolve_upgrade_pip_command(commands)
+        if upgrade_pip_command:
+            commands.insert(0, upgrade_pip_command)
+
```

Comment:
this can be problematic, as the user might want to install stuff that are required to be run first and now u inject that to be run first, I think this should be added before mlrun and not before all commands